### DISCUSSION TITLE
CodeQL on PR question ?

### DISCUSSION THREAD

**Discussion by aleks-ivanov** [microsoft/RulesEngine#123] [Q&A]
Why do you use the code from the head of the pull request for the CodeQL scanning ?

https://github.com/microsoft/RulesEngine/blob/f3ac4316dfde9baa9a7d2586f6c7a0fd2d0813d7/.github/workflows/codeql-analysis.yml#L26

**Comment by abbasc52** [comment]
hi @aleks-ivanov , thanks for pointing out. The action file was auto-generated, so we didn't look much into it. 
Also looks like the latest auto-generated codeql workflow does not have the above condition so we will go ahead fix this.
